src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -0,0 +1,359 @@
+    ) -> TextGenerationOutput:
+        """
+        Convert the engine outputs to the output schema for the pipeline.
+
+        :param engine_outputs: the outputs from the engine
+        :return: the output schema for the pipeline
+        """
+        generated_tokens, generated_logits = engine_outputs
+        # TODO: Make sure it works with batch size > 1
+        sequence = self.tokenizer.decode(
+            generated_tokens[0][0], skip_special_tokens=True
+        )
+        logits = generated_logits if kwargs.get("return_logits") else None
+
+        return TextGenerationOutput(sequence=sequence, logits=logits)
+
+    def engine_forward(
+        self, engine_inputs: List[numpy.ndarray], **kwargs
+    ) -> Tuple[numpy.ndarray, numpy.ndarray]:
+        """
+        Run the forward pass on the engine.
+
+        :param engine_inputs: list of numpy inputs to
+            Pipeline engine forward pass
+        :return: A tuple of numpy array that contains the
+            sequence of generated tokens and a sequence
+            of logits for each generated token
+        """
+        # run the prompt through
+        tokens, logits = self.prompt_inference(engine_inputs)
+        num_prompt_tokens = len(tokens) - 1
+
+        # create the generated output
+        max_tokens = (
+            self.max_generated_tokens
+            if self.max_generated_tokens and self.max_generated_tokens > 0
+            else 100 * self.sequence_length
+        )  # set safety for absolute max generation
+
+        generated_tokens = [tokens[-1]]
+        generated_logits = [logits]
+
+        while len(generated_tokens) < max_tokens:
+            (
+                token,
+                logits,
+            ) = self.autoregressive_inference(tokens, num_prompt_tokens)
+            tokens.append(token)
+            generated_tokens.append(token)
+            generated_logits.append(logits)
+
+            if token == self.tokenizer.eos_token_id and not self.force_max_tokens:
+                break
+
+        return numpy.array([[generated_tokens]]), numpy.concatenate(
+            generated_logits, axis=1
+        )
+
+    def prompt_inference(
+        self, engine_inputs: List[numpy.ndarray]
+    ) -> Tuple[List[int], Dict[str, numpy.ndarray]]:
+        """
+        An inference run that processes the prompt through the
+        model to generate the new token and logits
+
+        :param engine_inputs: the prompt (context) represented by a
+            list of numpy inputs to the engine
+        :return: A tuple of:
+            - The list of prompt tokens plus the new, generated token
+            - The logits generated from the prompt (with dimensions
+            ['batch_size', 'num_tokens', 'vocab_size'])
+        """
+        # get tokens by attention mask
+        tokens = engine_inputs[0][engine_inputs[1].nonzero()].tolist()
+        new_token = None
+
+        if (
+            self.prompt_batch_threshold is None
```

Comment:
total length prompt / total_length max length that we can support --> threshold

remove this treshold, replace it with the multitoken sequence length (by default 64)